### ISSUE TITLE
Function with long processing gets idle after 2 minutes and timeouts (Firebase Emulator only)

### ISSUE THREAD

**Issue by Corchoneitor** [firebase/firebase-tools#4378]
### Environment info

**firebase-tools:** 10.1.5

**Platform:** Windows

**Node:** 12.21.0

### Test case

Problem occurs only in Firebase emulator with functions that take more than 120 seconds of processing.
I have an onCall function which I run with **timeoutSeconds 180 seconds**. In this function, I do processing for over 130 seconds.
After the process is done, the function doesn't return any result and also doesn't finish. Therefore, it gets "idle" until the timeout (180 seconds) is reached and then throws the timeout error. Then the user who did the call doesn't get any response.

### Steps to reproduce

```
exports.setTestEnvironment = functions
	.runWith({
		timeoutSeconds: 180,
	})
	.https.onCall(async (data: any, context: any) => {
		try {
			await new Promise(resolve => setTimeout(resolve, 130000)); // // Sleep for 130 seconds, simulating processing for test purposes.
			console.log('Done. Should return success and finish execution');

			return {
				success: true,
			};
		} catch (error) {
			throw new functions.https.HttpsError('invalid-argument', error.message);
		}
	});
```

### Expected behavior

![image](https://user-images.githubusercontent.com/2912593/160896573-75bf3ece-0968-4c16-bab3-81091ad3bb83.png)

And the json response to the user...

```
{
  "result": {
    "success": true
  }
}
```

### Actual behavior

![image](https://user-images.githubusercontent.com/2912593/160897177-385a9c41-20dc-41ce-915e-1b440bbe7037.png)

User gets "ERR_EMPTY_RESPONSE".

This error only happens with the Firebase Emulator, and I'm not able to test my app correctly. The function works fine when it's deployed.
I have not encountered this issue in github nor stackoverflow, which should be very straight forward to get.

Feel free to ask any question or present any workaround.

Thanks.

**Comment by fvulich** [comment]
I experience the same issue with "firebase-tools": "9.3.0",

**Comment by Corchoneitor** [comment]
Any update on this? Thanks.

**Comment by Corchoneitor** [comment]
For other needs, I upgraded these dependencies and now the problem is solved.

node: 14.18.0
firebase-tools: 11.8.0
firebase-functions: 3.22.0

**Comment by joehan** [comment]
Fixed by #5464
